File: stack/71.py
class Solution:
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        arr = path.split('/')
        stack = []
        for s in arr:
            if s is '' or s == '.':
                continue
            if s == '..':
                try:
                    stack.pop()
                except IndexError:
                    continue
            else:
                stack.append(s)
                
        if len(stack) == 0:
            return '/'
        ret = ''
        while len(stack) > 0:
            ret = '/' + stack.pop() + ret
        return ret
                
        

File: stack/71.cc
class Solution {
public:
    string simplifyPath(string path) {
        std::stack<std::string> s;
        std::string temp;
        bool in = false;
        for (const auto& c : path) {
            if (c == '/') {
                if (!temp.empty()) {
                    if (temp == "..") {
                        if (!s.empty()) s.pop();
                    } else if (temp != ".") {
                        s.emplace(temp);
                    } 
                    temp = "";
                }   
            } else {
                temp += c;
            } 
        }
        if (!temp.empty()) {
            if (temp == "..") {
                if (!s.empty()) s.pop();
            } else if (temp != ".") {
                s.emplace(temp);
            } 
            temp = "";
        }  
        if (s.empty()) return "/";
        std::string ret;
        while (!s.empty()) {
            ret = "/"+ s.top() + ret;
            s.pop();
        }
        return ret;
    }
};
